### DISCUSSION TITLE
Can't open Signal's sqlite file

### DISCUSSION THREAD

**Discussion by thephm** [sqlitebrowser/sqlitebrowser#3717] [Q&A] [answered]
I created a tool that converts the `messages` table from Signal into Markdown: https://github.com/thephm/signal_sqlite_md

In that link you can see I describe the steps to get the `db.sqlite` key and how to open the file in DB4S using Raw and preprending `0x`

This whole process worked well until this month when I can no longer open the `db.sqlite` file in DB4S. I've tried 6 different versions of DB4S and when I try to open the file the dialog goes away and quickly comes back but the database is not opened.

I'm at a loss as to what to do next. Any ideas? Thanks in advance!

**Comment by justinclift** [comment]
Do you have a SQLite file from before the problem started happening?

Am kind of thinking it sounds like maybe the on-disk format or encryption options being used by the Signal application may have changed.

If an older SQLite file still opens, that would confirm the difference is in the file format *in some way*.  If it doesn't though, then it sounds like something has changed with your system in an unexpected way?

**Reply by justinclift** [reply]
Ahhh.  It looks like you've been using a nightly build of 3.13.x successfully for a while, and the recent (?) nightly builds are what's stopped working.  That does potentially sound like something we changed could have mucked things up then.

Do you still have any of the older builds around, and do they work?  If we can narrow down exactly when it stopped working, then that'll massive help us figure out what change we made broke things.

If it helps, we have an archive of nightly builds here:

https://nightlies.sqlitebrowser.org  <-- pick the sub-directory that matches the builds you use.  ie. win32 or win64

**Comment by justinclift** [comment, marked as answer]
As a different thought, how long have you been using this process?

Asking because the `--showdesktopkey` option seems like it was only added to the signal backup tools about a week ago?

https://github.com/bepaald/signalbackup-tools/issues/234#issuecomment-2308722897

**Reply by thephm** [reply]
I figured it out last night sorry and we found the same thing. Due to https://www.bleepingcomputer.com/news/security/signal-downplays-encryption-key-flaw-fixes-it-after-x-drama/ they changed `Key` to `EncryptedKey` in their config file. So sorry to have wasted your time. Using that `--showdesktopkey` gave the unencrypted key and it worked! I should've updated the ticket instead of going to bed.

**Reply by justinclift** [reply]
No worries at all, glad it turned out to be an easy solution. :smile:
